pkg/apis/pipeline/v1alpha1/build_gcs_resource.go

```diff
@@ -134,23 +126,37 @@ func (s *BuildGCSResource) Replacements() map[string]string {
 	}
 }
 
-// GetDownloadSteps returns the Steps needed to populate the workspace with the
-// resource's data.
-func (s *BuildGCSResource) GetDownloadSteps(sourcePath string) ([]Step, error) {
+// GetInputTaskModifier returns a TaskModifier that prepends a step to a Task to fetch the archive or manifest.
+func (s *BuildGCSResource) GetInputTaskModifier(ts *TaskSpec, sourcePath string) (TaskModifier, error) {
```

Comment:
im a bit iffy about `ts` being in this interface, what do you think @dlorenc ? From what I can tell it seems to be for the sake of the `getStorageVolumeSpec` function, and it looks like that's being used for:

```
func getStorageVolumeSpec(s PipelineStorageResourceInterface, spec *TaskSpec) ([]corev1.Volume, error) {
	var storageVol []corev1.Volume
	mountedSecrets := map[string]string{}

	for _, volume := range spec.Volumes {
		mountedSecrets[volume.Name] = ""
	}
...
}
```

I'm having a hard time understanding what that's actually doing and why it wants access to the TaskSpec to do it

Reply:
It looks like it's needed to prevent adding a volume to a task with a name that already exists. I'm not sure if that is actually a realistic scenario, or being overly defensive.

At a minimum, I'll change this to just pass ts as an object rather than a pointer so it can't be mutated.